### ISSUE TITLE
multiple bases have instance lay-out conflict on HumanMessageChunk  class on langchain 0.0.249

### ISSUE THREAD

**Issue by viduz** [langchain-ai/langchain#8577]
### Other notes 
Using this on notebook in Azure Synapse Studio

### Error 
```
When import happening getting bellow error message 
TypeError                                 Traceback (most recent call last)
/tmp/ipykernel_17180/722847199.py in <module>
----> 1 from langchain.vectorstores import Pinecone
      2 from langchain.embeddings.openai import OpenAIEmbeddings
      3 
      4 import datetime
      5 from datetime import date, timedelta

~/cluster-env/clonedenv/lib/python3.8/site-packages/langchain/__init__.py in <module>
      4 from typing import Optional
      5 
----> 6 from langchain.agents import MRKLChain, ReActChain, SelfAskWithSearchChain
      7 from langchain.cache import BaseCache
      8 from langchain.chains import (

~/cluster-env/clonedenv/lib/python3.8/site-packages/langchain/agents/__init__.py in <module>
      1 """Interface for agents."""
----> 2 from langchain.agents.agent import (
      3     Agent,
      4     AgentExecutor,
      5     AgentOutputParser,

~/cluster-env/clonedenv/lib/python3.8/site-packages/langchain/agents/agent.py in <module>
     13 from pydantic import BaseModel, root_validator
     14 
---> 15 from langchain.agents.agent_iterator import AgentExecutorIterator
     16 from langchain.agents.agent_types import AgentType
     17 from langchain.agents.tools import InvalidTool

~/cluster-env/clonedenv/lib/python3.8/site-packages/langchain/agents/agent_iterator.py in <module>
     19 )
     20 
---> 21 from langchain.callbacks.manager import (
     22     AsyncCallbackManager,
     23     AsyncCallbackManagerForChainRun,

~/cluster-env/clonedenv/lib/python3.8/site-packages/langchain/callbacks/__init__.py in <module>
      1 """Callback handlers that allow listening to events in LangChain."""
      2 
----> 3 from langchain.callbacks.aim_callback import AimCallbackHandler
      4 from langchain.callbacks.argilla_callback import ArgillaCallbackHandler
      5 from langchain.callbacks.arize_callback import ArizeCallbackHandler

~/cluster-env/clonedenv/lib/python3.8/site-packages/langchain/callbacks/aim_callback.py in <module>
      3 
      4 from langchain.callbacks.base import BaseCallbackHandler
----> 5 from langchain.schema import AgentAction, AgentFinish, LLMResult
      6 
      7 

~/cluster-env/clonedenv/lib/python3.8/site-packages/langchain/schema/__init__.py in <module>
      1 from langchain.schema.agent import AgentAction, AgentFinish
      2 from langchain.schema.document import BaseDocumentTransformer, Document
----> 3 from langchain.schema.memory import BaseChatMessageHistory, BaseMemory
      4 from langchain.schema.messages import (
      5     AIMessage,

~/cluster-env/clonedenv/lib/python3.8/site-packages/langchain/schema/memory.py in <module>
      5 
      6 from langchain.load.serializable import Serializable
----> 7 from langchain.schema.messages import AIMessage, BaseMessage, HumanMessage
      8 
      9 

~/cluster-env/clonedenv/lib/python3.8/site-packages/langchain/schema/messages.py in <module>
    136 
    137 
--> 138 class HumanMessageChunk(HumanMessage, BaseMessageChunk):
    139     pass
    140 

~/cluster-env/clonedenv/lib/python3.8/site-packages/pydantic/main.cpython-38-x86_64-linux-gnu.so in pydantic.main.ModelMetaclass.__new__()

~/cluster-env/clonedenv/lib/python3.8/abc.py in __new__(mcls, name, bases, namespace, **kwargs)
     83         """
     84         def __new__(mcls, name, bases, namespace, **kwargs):
---> 85             cls = super().__new__(mcls, name, bases, namespace, **kwargs)
     86             _abc_init(cls)
     87             return cls

TypeError: multiple bases have instance lay-out conflict
```

### Other installed packages in the system 

Package                       Version
----------------------------- -------------------
absl-py                       0.13.0
adal                          1.2.7
adlfs                         0.7.7
aiohttp                       3.8.5
aiosignal                     1.3.1
annotated-types               0.5.0
appdirs                       1.4.4
applicationinsights           0.11.10
argon2-cffi                   21.3.0
argon2-cffi-bindings          21.2.0
astor                         0.8.1
astunparse                    1.6.3
async-timeout                 4.0.2
attrs                         21.2.0
azure-common                  1.1.27
azure-core                    1.16.0
azure-datalake-store          0.0.51
azure-graphrbac               0.61.1
azure-identity                1.4.1
azure-mgmt-authorization      0.61.0
azure-mgmt-containerregistry  8.0.0
azure-mgmt-core               1.3.0
azure-mgmt-keyvault           2.2.0
azure-mgmt-resource           13.0.0
azure-mgmt-storage            11.2.0
azure-storage-blob            12.8.1
azure-synapse-ml-predict      1.0.0
azureml-core                  1.34.0
azureml-dataprep              2.22.2
azureml-dataprep-native       38.0.0
azureml-dataprep-rslex        1.20.2
azureml-dataset-runtime       1.34.0
azureml-mlflow                1.34.0
azureml-opendatasets          1.34.0
azureml-synapse               0.0.1
azureml-telemetry             1.34.0
backcall                      0.2.0
backports.functools-lru-cache 1.6.4
backports.tempfile            1.0
backports.weakref             1.0.post1
beautifulsoup4                4.9.3
bleach                        5.0.1
blinker                       1.4
bokeh                         2.3.2
Brotli                        1.0.9
brotlipy                      0.7.0
cachetools                    4.2.2
certifi                       2021.5.30
cffi                          1.14.5
chardet                       4.0.0
charset-normalizer            3.2.0
click                         8.0.1
cloudpickle                   1.6.0
conda-package-handling        1.7.3
configparser                  5.0.2
contextlib2                   0.6.0.post1
cryptography                  3.4.7
cycler                        0.10.0
Cython                        0.29.23
cytoolz                       0.11.0
dash                          1.20.0
dash-core-components          1.16.0
dash-cytoscape                0.2.0
dash-html-components          1.1.3
dash-renderer                 1.9.1
dash-table                    4.11.3
dask                          2021.6.2
databricks-cli                0.12.1
dataclasses-json              0.5.14
debugpy                       1.3.0
decorator                     4.4.2
defusedxml                    0.7.1
dill                          0.3.4
distlib                       0.3.6
distro                        1.7.0
dnspython                     2.4.1
docker                        4.4.4
dotnetcore2                   2.1.23
entrypoints                   0.3
et-xmlfile                    1.1.0
fastjsonschema                2.16.1
filelock                      3.8.0
fire                          0.4.0
Flask                         2.0.1
Flask-Compress                0.0.0
flatbuffers                   1.12
frozenlist                    1.4.0
fsspec                        2021.10.0
fsspec-wrapper                0.1.6
fusepy                        3.0.1
future                        0.18.2
gast                          0.3.3
gensim                        3.8.3
geographiclib                 1.52
geopy                         2.1.0
gevent                        21.1.2
gitdb                         4.0.7
GitPython                     3.1.18
google-auth                   1.32.1
google-auth-oauthlib          0.4.1
google-pasta                  0.2.0
greenlet                      1.1.0
grpcio                        1.37.1
h5py                          2.10.0
html5lib                      1.1
hummingbird-ml                0.4.0
idna                          2.10
imagecodecs                   2021.3.31
imageio                       2.9.0
importlib-metadata            4.6.1
importlib-resources           5.9.0
ipykernel                     6.0.1
ipython                       7.23.1
ipython-genutils              0.2.0
ipywidgets                    7.6.3
isodate                       0.6.0
itsdangerous                  2.0.1
jdcal                         1.4.1
jedi                          0.18.0
jeepney                       0.6.0
Jinja2                        3.0.1
jmespath                      0.10.0
joblib                        1.0.1
jsonpickle                    2.0.0
jsonschema                    4.15.0
jupyter-client                6.1.12
jupyter-core                  4.7.1
jupyterlab-pygments           0.2.2
jupyterlab-widgets            3.0.3
Keras-Applications            1.0.8
Keras-Preprocessing           1.1.2
keras2onnx                    1.6.5
kiwisolver                    1.3.1
koalas                        1.8.0
KqlmagicCustom                0.1.114.post8
langchain                     0.0.249
langsmith                     0.0.16
liac-arff                     2.5.0
library-metadata-cooker       0.0.7
lightgbm                      3.2.1
lime                          0.2.0.1
llvmlite                      0.36.0
locket                        0.2.1
loguru                        0.7.0
lxml                          4.6.5
Markdown                      3.3.4
MarkupSafe                    2.0.1
marshmallow                   3.20.1
matplotlib                    3.4.2
matplotlib-inline             0.1.2
mistune                       2.0.4
mleap                         0.17.0
mlflow-skinny                 1.18.0
msal                          1.12.0
msal-extensions               0.2.2
msrest                        0.6.21
msrestazure                   0.6.4
multidict                     5.1.0
mypy                          0.780
mypy-extensions               0.4.3
nbclient                      0.6.7
nbconvert                     7.0.0
nbformat                      5.4.0
ndg-httpsclient               0.5.1
nest-asyncio                  1.5.5
networkx                      2.5.1
nltk                          3.6.2
notebook                      6.4.12
notebookutils                 3.1.2-20230518.1
numba                         0.53.1
numexpr                       2.8.4
numpy                         1.24.4
oauthlib                      3.1.1
olefile                       0.46
onnx                          1.9.0
onnxconverter-common          1.7.0
onnxmltools                   1.7.0
onnxruntime                   1.7.2
openai                        0.27.8
openapi-schema-pydantic       1.2.4
openpyxl                      3.0.7
opt-einsum                    3.3.0
packaging                     21.0
pandas                        1.2.3
pandasql                      0.7.3
pandocfilters                 1.5.0
parso                         0.8.2
partd                         1.2.0
pathspec                      0.8.1
patsy                         0.5.1
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        8.2.0
pinecone-client               2.2.2
pip                           23.2.1
pkgutil_resolve_name          1.3.10
platformdirs                  2.5.2
plotly                        4.14.3
pmdarima                      1.8.2
pooch                         1.4.0
portalocker                   1.7.1
prettytable                   2.4.0
prometheus-client             0.14.1
prompt-toolkit                3.0.19
protobuf                      3.15.8
psutil                        5.8.0
ptyprocess                    0.7.0
py4j                          0.10.9
pyarrow                       3.0.0
pyasn1                        0.4.8
pyasn1-modules                0.2.8
pycairo                       1.20.1
pycosat                       0.6.3
pycparser                     2.20
pydantic                      1.8.2
pydantic_core                 2.4.0
Pygments                      2.9.0
PyGObject                     3.40.1
PyJWT                         2.1.0
pyodbc                        4.0.30
pyOpenSSL                     20.0.1
pyparsing                     2.4.7
pyperclip                     1.8.2
PyQt5                         5.12.3
PyQt5_sip                     4.19.18
PyQtChart                     5.12
PyQtWebEngine                 5.12.1
pyrsistent                    0.18.1
PySocks                       1.7.1
pyspark                       3.1.2
python-dateutil               2.8.1
pytz                          2021.1
pyu2f                         0.1.5
PyWavelets                    1.1.1
PyYAML                        5.4.1
pyzmq                         22.1.0
regex                         2023.6.3
requests                      2.31.0
requests-oauthlib             1.3.0
retrying                      1.3.3
rsa                           4.7.2
ruamel.yaml                   0.17.4
ruamel.yaml.clib              0.2.6
ruamel-yaml-conda             0.15.100
SALib                         1.3.11
scikit-image                  0.18.1
scikit-learn                  0.23.2
scipy                         1.5.3
seaborn                       0.11.1
SecretStorage                 3.3.1
Send2Trash                    1.8.0
setuptools                    49.6.0.post20210108
shap                          0.39.0
six                           1.16.0
skl2onnx                      1.8.0
sklearn-pandas                2.2.0
slicer                        0.0.7
smart-open                    5.1.0
smmap                         3.0.5
soupsieve                     2.2.1
SQLAlchemy                    1.4.20
sqlanalyticsconnectorpy       1.0.1
statsmodels                   0.12.2
synapseml-cognitive           0.10.2.dev1
synapseml-core                0.10.2.dev1
synapseml-deep-learning       0.10.2.dev1
synapseml-internal            0.0.0.dev1
synapseml-lightgbm            0.10.2.dev1
synapseml-opencv              0.10.2.dev1
synapseml-vw                  0.10.2.dev1
tabulate                      0.8.9
tenacity                      8.2.2
tensorboard                   2.4.1
tensorboard-plugin-wit        1.8.0
tensorflow                    2.4.1
tensorflow-estimator          2.4.0
termcolor                     1.1.0
terminado                     0.15.0
textblob                      0.15.3
threadpoolctl                 2.1.0
tifffile                      2021.4.8
tiktoken                      0.4.0
tinycss2                      1.1.1
toolz                         0.11.1
torch                         1.8.1
torchvision                   0.9.1
tornado                       6.1
tqdm                          4.65.0
traitlets                     5.0.5
typed-ast                     1.4.3
typing_extensions             4.5.0
typing-inspect                0.8.0
urllib3                       1.26.4
virtualenv                    20.14.0
wcwidth                       0.2.5
webencodings                  0.5.1
websocket-client              1.1.0
Werkzeug                      2.0.1
wheel                         0.36.2
widgetsnbextension            3.5.2
wrapt                         1.12.1
xgboost                       1.4.0
XlsxWriter                    3.0.3
yarl                          1.6.3
zipp                          3.5.0
zope.event                    4.5.0
zope.interface                5.4.0

``

### Information

- [ ] The official example notebooks/scripts
- [X] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [X] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [X] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [X] Async

### Reproduction

Go to Azure Synapse Studio 
Open a Notebook
Select Pyspark(Python) as Language
Please note your node should run python 3.8

Then put bellow 

### Installed bellow to the session. 
```
!pip install --upgrade pip
!pip install tqdm
!pip install pinecone-client
!pip install typing-extensions==4.5.0
!pip install langchain
!pip install openai
!pip install tiktoken
```

### Here are my imports 
```
from langchain.vectorstores import Pinecone
from langchain.embeddings.openai import OpenAIEmbeddings

import datetime
from datetime import date, timedelta
import time
import csv

import openai
import requests
import pandas as pd
from pyspark.sql import SparkSession
from pyspark.sql.functions import col, lit, current_timestamp
from pyspark.sql import functions as F
from pyspark.sql import Row
from pyspark.sql.utils import AnalysisException
from pyspark.sql.types import StructType, StructField, StringType, TimestampType, LongType
from pyspark.sql.window import Window
from pyspark.sql import types as T
import concurrent.futures
import pinecone
import tiktoken
```

Run the notebook.

### Expected behavior

Import should be happend without any issues.

**Comment by viduz** [comment]
Hi Team, 
Latest update on this, I have cleared out noises around this and finally. Isolated issues to these two imports. 

install Langchian package. 

`!pip install langchain`

```
from langchain.vectorstores import Pinecone
from langchain.embeddings.openai import OpenAIEmbeddings
```

Then I'm getting bellow issue 

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
/tmp/ipykernel_6996/3251986536.py in <module>
----> 1 from langchain.vectorstores import Pinecone
      2 from langchain.embeddings.openai import OpenAIEmbeddings

~/cluster-env/clonedenv/lib/python3.8/site-packages/langchain/__init__.py in <module>
      4 from typing import Optional
      5 
----> 6 from langchain.agents import MRKLChain, ReActChain, SelfAskWithSearchChain
      7 from langchain.cache import BaseCache
      8 from langchain.chains import (

~/cluster-env/clonedenv/lib/python3.8/site-packages/langchain/agents/__init__.py in <module>
     29 
     30 """  # noqa: E501
---> 31 from langchain.agents.agent import (
     32     Agent,
     33     AgentExecutor,

~/cluster-env/clonedenv/lib/python3.8/site-packages/langchain/agents/agent.py in <module>
     11 
     12 import yaml
---> 13 from pydantic import BaseModel, root_validator
     14 
     15 from langchain.agents.agent_iterator import AgentExecutorIterator

~/cluster-env/clonedenv/lib/python3.8/site-packages/pydantic/__init__.cpython-38-x86_64-linux-gnu.so in init pydantic.__init__()

~/cluster-env/clonedenv/lib/python3.8/site-packages/pydantic/dataclasses.cpython-38-x86_64-linux-gnu.so in init pydantic.dataclasses()

ImportError: cannot import name dataclass_transform

**Comment by buckmaxwell** [comment]
I am seeing this in 0.251 as well

**Comment by dunnkevin** [comment]
Was also seeing this - Was able to work around it by upgrading Pydantic 1.8.2 -> 1.10.12.

**Comment by zky001** [comment]
> Was also seeing this - Was able to work around it by upgrading Pydantic 1.8.2 -> 1.10.12.

good job

**Comment by aadhil96** [comment]
I am also getting same error 

`TypeError: multiple bases have instance lay-out conflict`

How to solve it ?

**Comment by gk2020-coder** [comment]
Collecting jinja
  Using cached Jinja-1.2.tar.gz (252 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [7 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-snyj1guu/jinja_5c60781821294149bc3e6c5dcfa1f808/setup.py", line 28
          except DistutilsError, e:
                 ^^^^^^^^^^^^^^^^^
      SyntaxError: multiple exception types must be parenthesized
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

**Comment by PavelAgurov** [comment]
I have the same error today

**Comment by eastonsuo** [comment]
I meet this error when use jupyter notebook in **pycharm**.
but it does not occur when i use jupyter lab in **terminal**

**Comment by Pratipkhandelwal** [comment]
Having the same issue in VS terminal, the same was working a week back. Any solution here?

**Comment by Fiztech-ai** [comment]
Same issure here, upgrading Pydantic 1.8.2 -> 1.10.12 does not help.

**Comment by suleymanov** [comment]
I'm facing the same error, could isolate it with instruction `from langchain.schema.messages import BaseMessage`. As far as I can inspect it then goes into file `langchain.schema.__init__.py`, in which tries to execute first row `from langchain.schema.agent import AgentAction, AgentFinish`, goes into the file `langchain.schema.agent` and there tries to import `BaseMessage` again which in my understanding should cause an infinite loop... I'm inspecting version **0.0.322** which corresponds to commit *4d6243fa879fddef0e85904a3538af715da627e8*. And the error "multiple bases have instance lay-out conflict" arises only in jupyter notebook, when I run the same code in terminal everything is fine.

**Comment by rick-mooney** [comment]
I had a similar issue popup as well, but found it wasnt specific to importing a specific module from langchain - it was just importing anything from langchain.  I uninstalled my env and reinstalled everything and that solved the issue.

**Comment by vvanglro** [comment]
Looks like it has something to do with this: https://github.com/pydantic/pydantic/issues/3523

**Comment by uchiiii** [comment]
In my case, it is solved by `pip install pydantic==2.5.2`. (Sorry if it does not help though)

**Comment by Ezhil6999** [comment]
My error resolved by upgrading pydantic,

pip install -U pydantic
pip install --upgrade pydantic
pip install pydantic

**Comment by yesidc** [comment]
having the same issue
`langchain           0.1.0`

**Comment by benm5678** [comment]
I get the same in slightly different spot:
```
  File "/Users/benm/Library/Python/3.11/lib/python/site-packages/langchain_core/messages/ai.py", line 29, in <module>
    class AIMessageChunk(AIMessage, BaseMessageChunk):
  File "/usr/local/lib/python3.11/site-packages/pydantic/main.py", line 367, in __new__
    cls = super().__new__(mcs, name, bases, new_namespace, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen abc>", line 106, in __new__
TypeError: multiple bases have instance lay-out conflict
```

Using latest:
```
Name: langchain
Version: 0.1.7

Name: pydantic
Version: 2.6.1
```

**Comment by benm5678** [comment]
> I get the same in slightly different spot:
> 
> ```
>   File "/Users/benm/Library/Python/3.11/lib/python/site-packages/langchain_core/messages/ai.py", line 29, in <module>
>     class AIMessageChunk(AIMessage, BaseMessageChunk):
>   File "/usr/local/lib/python3.11/site-packages/pydantic/main.py", line 367, in __new__
>     cls = super().__new__(mcs, name, bases, new_namespace, **kwargs)
>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "<frozen abc>", line 106, in __new__
> TypeError: multiple bases have instance lay-out conflict
> ```
> 
> Using latest:
> 
> ```
> Name: langchain
> Version: 0.1.7
> 
> Name: pydantic
> Version: 2.6.1
> ```

Ah...in my case, I was running it via pyhttpdbg or streamlit, and those were launched from different python env.  So paying attention to paths that show up in error can help you verify it's running on expected venv.

**Comment by bharatchandra-dev** [comment]
> In my case, it is solved by `pip install pydantic==2.5.2`. (Sorry if it does not help though)

This solved my problem. Thank you
